src/client/components/app/Navigation/index.tsx

```diff
@@ -19,49 +19,52 @@ export interface NavigationLink {
 
 const StyledNavigation = styled.div``;
 
-const navLinks: NavigationLink[] = [
-  {
-    to: '/portfolio',
-    text: 'navigation.portfolio',
-    icon: WalletIcon,
-  },
-  {
-    to: '/market',
-    text: 'navigation.market',
-    icon: VaultIcon,
-  },
-  {
-    to: 'https://docs.debtdao.finance',
-    text: 'navigation.docs',
-    icon: SearchIcon,
-    external: true,
-    optional: true,
-  },
-  {
-    to: 'https://discord.gg/F83xx67fyQ',
-    text: 'navigation.discord',
-    icon: DiscordIcon,
-    external: true,
-    optional: true,
-  },
-  {
-    to: '/settings',
-    text: 'navigation.settings',
-    icon: SettingsIcon,
-    // hideMobile: true,
-  },
-];
-
 interface NavigationProps {
   hideOptionalLinks?: boolean;
 }
 
 export const Navigation = ({ hideOptionalLinks }: NavigationProps) => {
   const { isMobile, isTablet, isDesktop } = useWindowDimensions();
+  const userWallet = useAppSelector(WalletSelectors.selectSelectedAddress);
+  const dispatch = useAppDispatch();
+
+  const navLinks: NavigationLink[] = [
```

Comment:
keep definition outside of component

Reply:
Moved out of definition, with links that require dynamic state data I defined directly outside of map